### PR TITLE
docs: Document testing nuqs based hooks

### PR DESCRIPTION
**Pull request by sebastianvitterso**
See #794.

### CODE REVIEW

**Review by franky47 — APPROVED** [review]
Thanks, this could even be added at the top before each testing framework's example, to introduce the concept of the `withNuqsTestingAdapter` factory function beforehand.

**Comment by sebastianvitterso** [comment]
I've allowed edits by maintainers, so feel free to adjust the ordering of the document!

**Comment by sebastianvitterso** [comment]
I guess you'll merge when you want :v:

**Review by franky47 — APPROVED** [review]

File: packages/docs/content/docs/testing.mdx [hunk 1]
```diff
@@ -74,15 +92,15 @@ it('should increment the count when clicked', async () => {
...
+## Jest and ESM
 
 Since nuqs 2 is an [ESM-only package](https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c),
 there are a few hoops you need to jump through to make it work with Jest.
 This is extracted from the [Jest ESM guide](https://jestjs.io/docs/ecmascript-modules).
 
 1. Add the following options to your jest.config.ts file:
 
-```ts title="jest.config.ts"
+```ts title="jest.configx.ts"
```
Comment (sebastianvitterso) [inline packages/docs/content/docs/testing.mdx:103]: `configx`?

Comment (franky47) [reply packages/docs/content/docs/testing.mdx:103]: Good catch, thanks!

**Pull request merged**